### PR TITLE
Add a get IRC27NFT Metadata guide

### LINKED ISSUE #1571: Document `getIRC27NFTData` function
**Describe the solution you'd like**

As we found out in #1565 we need to create new docs for [`getIRC27NFTData`](https://github.com/iotaledger/wasp/blob/6efe78da8ad661701a154d3e0ec534d0c9244a3b/packages/vm/core/evm/iscmagic/ISCSandbox.sol#L118), which returns [`IRC27NFT`](https://github.com/iotaledger/wasp/blob/6efe78da8ad661701a154d3e0ec534d0c9244a3b/packages/vm/core/evm/iscmagic/ISCTypes.sol#L72-L75). 

We should create a how-to which explains how to use this function and especially how to encode the uri data in [`IRC27NFTMetadata `](https://github.com/iotaledger/wasp/blob/6efe78da8ad661701a154d3e0ec534d0c9244a3b/packages/vm/core/evm/iscmagic/ISCTypes.sol#L57-L69)

### PR DESCRIPTION
**Pull request by Ginowine**
# Description of change

This PR creates a new page titled "Get NFT Metadata". The page explains how to utilize the getIRC27NFTData function within a smart contract to fetch information about a specific IRC27 NFT on the IOTA Network

## Links to any relevant issues
fixes #1571 

## Type of change

- Documentation Enhancement

## Change checklist

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+tags:
+  - NFT
+  - EVM
+  - how-to
+---
+import GetNftMetadata from '../../../_partials/how-tos/token/_get-nft-metadata.md';
+
+# Get NFT Metadata
+
+This guide explains how to utilize the `getIRC27NFTData` function within a smart contract to fetch information about a specific IRC27 NFT on the IOTA Network. The function resides in the ISCSandbox contract and returns details pertaining to the requested NFT.
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md:13]:
```suggestion
This guide explains how to use the [`getIRC27NFTData`](../../../reference/magic-contract/ISCSandbox/#getirc27nftdata) function within a smart contract to fetch information about a specific IRC27 NFT on the IOTA Network. The function resides in the ISCSandbox contract and returns details pertaining to the requested NFT.
```

We could also cut the last sentence if we add the link to the refs.

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+## Function Signature:
+```solidity
+function getIRC27NFTData(bytes memory nftID) public view returns (IRC27NFT memory irc27NftData);
+```
+## Parameters:
+`NFTID id`: This parameter represents the unique identifier of the IRC27 NFT you intend to retrieve information about.
+## Return Value:
+The function returns a data structure of type IRC27NFT. This structure incorporates two elements:
+*  `nft`: This element provides details regarding the underlying on-chain NFT. It's of type `ISCNFT`.
+*  `metadata`: This element offers information specific to the IRC27 standard. It's of type `IRC27NFTMetadata`.
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md:28]:
````suggestion
### Function Signature:

```solidity
function getIRC27NFTData(bytes memory nftID) public view returns (IRC27NFT memory irc27NftData);
```

### Parameters:

`NFTID id`: This parameter represents the unique identifier of the IRC27 NFT you intend to retrieve information about.

### Return Value:

The function returns a data structure of type IRC27NFT. This structure incorporates two elements:
*  `nft`: This element provides details regarding the underlying on-chain NFT. It's of type `ISCNFT`.
*  `metadata`: This element offers information specific to the IRC27 standard. It's of type `IRC27NFTMetadata`.
````

This probably belongs in the refs, though I think they are autogenerated.

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+### Here's a step-by-step breakdown of how to leverage getIRC27NFTData to acquire IRC27 NFT metadata:
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md:30]:
```suggestion
## How To Use `getIRC27NFTData` 
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+1. Fetching IRC27 NFT Metadata:
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md:32]:
```suggestion
### 1. Fetch IRC27 NFT Metadata
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+Create a function called fetchNFTData in your contract that calls getIRC27NFTData and processes its return value.
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md:34]:
```suggestion
Create a function called `fetchNFTData` in your contract that calls `getIRC27NFTData` and processes its return value.
```
We usually wrap function names in ``

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+```solidity
+// Function to get NFT data
+    function fetchNFTData(bytes memory nftID) public view returns (IRC27NFT memory irc27NftData) {
+        bytes32 id = bytes32(nftID);
+        NFTID nftIDTyped = NFTID.wrap(id);
+        irc27NftData = ISC.sandbox.getIRC27NFTData(nftIDTyped);
+        return irc27NftData;
+    }
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md:43]:
```suggestion
function fetchNFTData(bytes memory nftID) public view returns (IRC27NFT memory irc27NftData) {
    bytes32 id = bytes32(nftID);
    NFTID nftIDTyped = NFTID.wrap(id);
    irc27NftData = ISC.sandbox.getIRC27NFTData(nftIDTyped);
    return irc27NftData;
}
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+```
+2. Encoding URI Data in IRC27NFTMetadata
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md:45]:
```suggestion
### 2. Encode the URI Data in `IRC27NFTMetadata`
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+The `IRC27NFTMetadata` struct holds detailed metadata, including the URI and other fields like name and description. Encoding this data correctly is crucial for ensuring the integrity and accessibility of the NFT's metadata.
+
+```solidity
+struct IRC27NFTMetadata {string uri; string name;string description;}
+```
+
+3. Implementing URI Encoding
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md:53]:
```suggestion
### 3. Implement URI Encoding
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+
+The OpenZeppelin Contracts library provides a `Base64` utility that simplifies encoding URI data. Here's how to implement URI encoding:
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md:55]:
```suggestion
The [OpenZeppelin Contracts](https://www.openzeppelin.com/) library provides a `Base64` utility that simplifies encoding URI data. Here's how to implement URI encoding:
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+
+## Implementation:
+
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md:58]:
```suggestion
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+        mimeType: _mimeType,
+        uri: encodedURI,
+        name: _name,
+        description: _description
+    });
+}
+```
+
+## Full Example Contract
+Combining all the above steps, here’s a complete example:
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md:94]:
```suggestion
## Full Example Contract

Combining all the above steps, here’s a complete example:
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+        name: _name
+    });
+}
+}
+```
+## Example Usage
+
+To mint a new NFT and fetch its metadata, you would use the above functions within a larger contract that handles NFT minting and management.
+
+<GetNftMetadata />
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md:153]:
```suggestion
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+  - NFT
+  - EVM
+  - how-to
+---
+import GetNftMetadata from '../../../_partials/how-tos/token/_get-nft-metadata.md';
+
+# Get NFT Metadata
+
+This guide explains how to utilize the `getIRC27NFTData` function within a smart contract to fetch information about a specific IRC27 NFT on the IOTA Network. The function resides in the ISCSandbox contract and returns details pertaining to the requested NFT.
+
```
Comment (lucas-tortora) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md:14]:
```suggestion


<GetNftMetadata />

```

**Review by lucas-tortora — CHANGES_REQUESTED** [review]
Very nice article :)  just some suggestions regarding structure

**Comment by Ginowine** [comment]
I have addressed all the review suggestions. Please take a look and let me know if there are any further changes needed.

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 1 above)
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: I have made the necessary updates based on your suggestions and also removed the last sentence as requested.

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 2 above)
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: I made the changes, but like you said it belongs in the refs, let me know if I should take this section out

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 3 above)
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: Thanks, done

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 4 above)
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: Done

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 5 above)
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: Thanks, noted.

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 6 above)
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: Done

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 7 above)
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: Done

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 8 above)
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: Thanks, it's updated

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 9 above)
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: Thanks, updated

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 10 above)
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: Done

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 12 above)
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: Done.

**Comment by Ginowine** [comment]
I noticed this URL (../../../reference/magic-contract/ISCSandbox/#getirc27nftdata) on the page is causing the test/build to fail. Trying to get the right URL. Will send a new PR after fixing it

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 1 above)
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: getIRC27NFTData url was broken, fixed it with this url: https://github.com/iotaledger/wasp/blob/develop/packages/vm/core/evm/iscmagic/ISCSandbox.sol#L118

**Review by lucas-tortora — APPROVED** [review]

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 1 above)
Comment (Dr-Electron) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: I will fix it. But we need the updated reference docs first 👍

**Review by Dr-Electron — CHANGES_REQUESTED** [review]
Really great article ❤️ 
I would like to change it a little bit tho.

Let me first try to explain the "issue".
When you send an NFT on L1 to the L1 account of the chain, it gets automatically registered as ERC721 token in our EVM. But as your IRC27 token is not fully compatible with the `tokenUri` of `ERC721` which returns a json, our EVM creates that json out of the `uri`, the `description` and the `name` of the IRC27 NFT on L1.

So normally people would never use the `getIRC27NFTData` function. The would just call the standard `tokenUri` of `ERC721`.

So with that in mind, I think the first thing we should do, is is adding an admonition that tells people to use tokenUri if possible. Only if the need some of the unique metadata of IRC27 they should use this (or technically if they develop something themselves to transfer this NFTs or whatever).

This also means we are more interested in decoding the metadata.uri.

Btw you can see how this function is used to get the tokenUri [here](https://github.com/iotaledger/wasp/blob/d96764b27c0cce58ffb4f94e52b2907aa81e4f8f/packages/vm/core/evm/iscmagic/ERC721NFTs.sol#L303-L307)

So TL:DR; :trollface: 

- [ ] Add admonition pointing people to https://wiki.iota.org/isc/how-tos/core-contracts/nft/use-as-erc721/ and telling them about the tokenUri
- [ ] Rewrite encoding to actually decode the JSON from metadata.uri.

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+image: /img/logo/WASP_logo_dark.png
+tags:
+  - NFT
+  - EVM
+  - how-to
+---
+import GetNftMetadata from '../../../_partials/how-tos/token/_get-nft-metadata.md';
+
+# Get NFT Metadata
+
```
Comment (Dr-Electron) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md:12]:
```suggestion

:::info ERC721

As your L1 NFT is always registered as [ERC721](https://eips.ethereum.org/EIPS/eip-721), you might want to get the metadata like `tokenURI` from there. Using `getIRC27NFTData` is normally only needed if you need special [IRC27](https://wiki.iota.org/tips/tips/TIP-0027/) metadata

```

File: docs/build/isc/v1.1/docs/_partials/how-tos/token/_get-nft-metadata.md [hunk 15]
```diff
@@ -0,0 +1,5 @@
+:::tip Mint an NFT 
+
+Mint your first NFT following our how to [mint an NFT guide](/isc/how-tos/core-contracts/nft/mint-nft/#about-nfts).
```
Comment (Dr-Electron) [inline docs/build/isc/v1.1/docs/_partials/how-tos/token/_get-nft-metadata.md:3]:
```suggestion
Mint your first NFT following our how to [mint an NFT guide](../how-tos/core-contracts/nft/mint-nft.md#about-nfts).
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 16]
```diff
@@ -0,0 +1,151 @@
...
+        mimeType: _mimeType,
+        uri: encodedURI,
+        name: _name
+    });
+}
+}
+```
+## Example Usage
+
+To mint a new NFT and fetch its metadata, you would use the above functions within a larger contract that handles NFT minting and management.
```
[the remark this answers is not in the document]
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: Ok thanks, updated the doc

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 17]
```diff
@@ -0,0 +1,151 @@
...
+### Parameters:
+
+`NFTID id`: This parameter represents the unique identifier of the IRC27 NFT you intend to retrieve information about.
+
+### Return Value:
+
+The function returns a data structure of type IRC27NFT. This structure incorporates two elements:
+*  `nft`: This element provides details regarding the underlying on-chain NFT. It's of type `ISCNFT`.
+*  `metadata`: This element offers information specific to the IRC27 standard. It's of type `IRC27NFTMetadata`.
+
```
[the remark this answers is not in the document]
Comment (Ginowine) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/core-contracts/nft/get-nft-metadata.md]: Thanks, deleted as suggested.

File: docs/build/isc/v1.1/docs/_partials/how-tos/token/_get-nft-metadata.md (hunk 15 above)
Comment (Ginowine) [reply docs/build/isc/v1.1/docs/_partials/how-tos/token/_get-nft-metadata.md]: Implemented, thanks.

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md (hunk 14 above)
Comment (Ginowine) [reply docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md:13]: Done, thanks

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 18]
```diff
@@ -0,0 +1,233 @@
...
+```solidity
+function fetchNFTData(bytes memory nftID) public view returns (IRC27NFT memory irc27NftData) {
+    bytes32 id = bytes32(nftID);
+    NFTID nftIDTyped = NFTID.wrap(id);
+    irc27NftData = ISC.sandbox.getIRC27NFTData(nftIDTyped);
+    return irc27NftData;
+}
+```
+### 2. Encode the URI Data in `IRC27NFTMetadata`
```
Comment (Dr-Electron) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md:38]: As agreed on Slack, let's remove the whole on-chain encode and decoding stuff

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 19]
```diff
@@ -0,0 +1,233 @@
...
+<GetNftMetadata />
+
+## Understanding the `getIRC27NFTData` Function
+
+The [`getIRC27NFTData`](../../../reference/magic-contract/ISCSandbox.md#getirc27nftdata) function retrieves metadata for an IRC27 NFT based on its identifier. IRC27 is a series of standards to support interoperable and universal NFT systems throughout the IOTA ecosystem.
+
+## How To Use `getIRC27NFTData`
+
+### 1. Fetch IRC27 NFT Metadata
+
```
Comment (Dr-Electron) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md:27]:
```suggestion
```



Not needed if we only have one title anymore

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 20]
```diff
@@ -0,0 +1,233 @@
...
+Create a function called `fetchNFTData` in your contract that calls `getIRC27NFTData` and processes its return value.
```
Comment (Dr-Electron) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md:28]:
```suggestion
Create a function called `fetchNFTData` in your contract that calls `getIRC27NFTData` and processes its return value. getIRC27NFTData returns a struct of type [`IRC27NFTMetadata`](../../../reference/magic-contract/ISCTypes.md#irc27nftmetadata) which contains properties like the NFT name, uri and more

:::info IRC27NFTMetadata URI

The uri property contains a JSON object which follows the `ERC721` standard. This JSON is also returned by the [`tokenURI`](../../../reference/magic-contract/ERC721NFTs.md#tokenuri) function from the `ERC721NFTs` contract.

:::
```

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 21]
```diff
@@ -0,0 +1,233 @@
+---
+description: How to get NFT metadata on L1
```
Comment (Dr-Electron) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md:2]:
```suggestion
description: How to get NFT metadata from a L1 NFT
```
```suggestion
description: How to get NFT metadata on L1
```

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md
Comment (Dr-Electron) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md:1]: This file should get the .mdx extension as it used mdx stuff.

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md [hunk 22]
```diff
@@ -0,0 +1,233 @@
...
+image: /img/logo/WASP_logo_dark.png
+tags:
+  - NFT
+  - EVM
+  - how-to
+---
+import GetNftMetadata from '../../../_partials/how-tos/token/_get-nft-metadata.md';
+import ERC721Admonition from '../../../_admonitions/_ERC721.md';
+
+# Get NFT Metadata
```
Comment (Dr-Electron) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.md:12]: Should we rename this to:
```suggestion
# Get IRC27 NFT Metadata
```
Would make it more clear that we are not talking about standard ERC721 or similar here

**Review by Dr-Electron — CHANGES_REQUESTED** [review]
As discussed on slack I think we can get rid of the encoding stuff. Sorry for realising that soo late.

Also you need to change version 1.3. In 1.1 some of this things are not available yet.

So I would move it to version 1.3 and we can think if it makes sense to add a stripped down version to 1.1, but I think just adding it to 1.3 should be fine.

**Comment by Ginowine** [comment]
Ok, thanks for the suggestions. I'll implement them and send a new push request.

**Comment by Ginowine** [comment]
@Dr-Electron, I have updated the doc to follow your suggestions. Please take a look and let me know. Thanks.

**Comment by lucas-tortora** [comment]
> @Dr-Electron, I have updated the doc to follow your suggestions. Please take a look and let me know. Thanks.

@Ginowine, could you fix the [broken links in the test workflow](https://github.com/iotaledger/iota-wiki/actions/runs/9458216267/job/26053359223?pr=1583#step:7:172)?

**Comment by Ginowine** [comment]
@lucas-tortora, I fixed the link pointing to the mint-nft.md file. For the other link errors below, Kevin said the preview doesn’t work because my fork doesn’t know iota-wiki deployment secrets and that we can ignore it as long as the build workflow succeeds.

`../../../reference/magic-contract/ERC721NFTs.md/#tokenuri (resolved as: /isc/v1.3-alpha/how-tos/reference/magic-contract/ERC721NFTs.md/)`

The below link was provided by Kevin.

```
../../../reference/magic-contract/ERC721NFTs.md#tokenuri
```
It would be great if I could get the correct link to use.

File: docs/build/isc/v1.1/docs/_admonitions/_IRC27.md [hunk 23]
```diff
@@ -0,0 +1,5 @@
+:::info IRC27NFTMetadata URI
+
+The uri property contains a JSON object which follows the `ERC721` standard. This JSON is also returned by the [`tokenURI`](../../../reference/magic-contract/ERC721NFTs.md#tokenuri) function from the `ERC721NFTs` contract.
```
Comment (lucas-tortora) [inline docs/build/isc/v1.1/docs/_admonitions/_IRC27.md:3]:
```suggestion
The uri property contains a JSON object which follows the `ERC721` standard. This JSON is also returned by the [`tokenURI`](../reference/magic-contract/ERC721NFTs.md#tokenuri) function from the `ERC721NFTs` contract.
```

File: docs/build/isc/v1.3-alpha/docs/_admonitions/_IRC27.md [hunk 24]
```diff
@@ -0,0 +1,5 @@
+:::info IRC27NFTMetadata URI
+
+The uri property contains a JSON object which follows the `ERC721` standard. This JSON is also returned by the [`tokenURI`](../../../reference/magic-contract/ERC721NFTs.md#tokenuri) function from the `ERC721NFTs` contract.
```
Comment (lucas-tortora) [inline docs/build/isc/v1.3-alpha/docs/_admonitions/_IRC27.md:3]:
```suggestion
The uri property contains a JSON object which follows the `ERC721` standard. This JSON is also returned by the [`tokenURI`](../reference/magic-contract/ERC721NFTs.md#tokenuri) function from the `ERC721NFTs` contract.
```

**Review by lucas-tortora — CHANGES_REQUESTED** [review]
this should fix the build :)

**Comment by Ginowine** [comment]
Thank you so much, @lucas-tortora, for the suggestions on how to fix the build. Noted.

**Review by lucas-tortora — APPROVED** [review]

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.mdx [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@iota/iscmagic/ISC.sol";
+import "@iota/iscmagic/ISCTypes.sol";
+
+contract IRCNFTMetadata {
+
+function fetchIRC27NFTData(uint256 tokenId) public view returns (IRC27NFT memory irc27NftData) {
+    irc27NftData = ISC.sandbox.getIRC27NFTData(tokenId.asNFTID());
```
Comment (salaheldinsoliman) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.mdx:52]: when trying this out, this line:
`irc27NftData = ISC.sandbox.getIRC27NFTData(tokenId.asNFTID());` didn't work,
But this: 
`irc27NftData = ISC.sandbox.getIRC27NFTData(ISCTypes.asNFTID(tokenId));`
worked, which is expected, since the `uint256` doesn't have a function called `asNFTID()`. 

This contradicts with this page in the wiki though:
https://wiki.iota.org/isc/how-tos/core-contracts/nft/use-as-erc721/ (in the `TOKEN ID TO NFT ID` section)

**Review by salaheldinsoliman — CHANGES_REQUESTED** [review]

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.mdx (hunk 25 above)
Comment (Ginowine) [reply docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-nft-metadata.mdx]: You are correct. The best implementation is using `irc27NftData = ISC.sandbox. getIRC27NFTData (ISCTypes. asNFTID (tokenId));`. Are you suggesting we update the `use-as-erc721` page to remove the use of `tokenID.asNFTID()`? An issue around `uint256` not having a function called `asNFTID()` might be a start.
